libV2/validationUtils.js

```diff
@@ -0,0 +1,2802 @@
+
+  // decide explodable params, starting value and separators between key-value and properties for serialisation
+  switch (style) {
+    case 'matrix':
+      isExplodable = paramSchema.type === 'object' || explode;
+      startValue = ';' + ((paramSchema.type === 'object' && explode) ? '' : (paramName + '='));
+      propSeparator = explode ? ';' : ',';
+      keyValueSeparator = explode ? '=' : ',';
+      break;
+    case 'label':
+      startValue = '.';
+      propSeparator = '.';
+      keyValueSeparator = explode ? '=' : '.';
+      break;
+    case 'form':
+      // for 'form' when explode is true, query is divided into different key-value pairs
+      propSeparator = keyValueSeparator = ',';
+      break;
+    case 'simple':
+      propSeparator = ',';
+      keyValueSeparator = explode ? '=' : ',';
+      break;
+    case 'spaceDelimited':
+      explode = false;
+      propSeparator = keyValueSeparator = '%20';
+      break;
+    case 'pipeDelimited':
+      explode = false;
+      propSeparator = keyValueSeparator = '|';
+      break;
+    case 'deepObject':
+      // for 'deepObject' query is divided into different key-value pairs
+      explode = true;
+      break;
+    default:
+      break;
+  }
+
+  return { style, explode, startValue, propSeparator, keyValueSeparator, isExplodable };
+}
+
+/**
+ * This function deserialises parameter value based on param schema
+ *
+ * @param {*} param - OpenAPI Parameter object
+ * @param {String} paramValue - Parameter value to be deserialised
+ * @param {String} parameterSource - Specifies whether the schema being faked is from a request or response.
+ * @param {Object} components - OpenAPI components defined in the OAS spec. These are used to
+ *  resolve references while generating params.
+ * @param {object} options - a standard list of options that's globally passed around. Check options.js for more.
+ * @param {Object} schemaCache - object storing schemaFaker and schemaResolution caches
+ * @returns {*} - deserialises parameter value
+ */
+function deserialiseParamValue (param, paramValue, parameterSource, components, options) {
+  var constructedValue,
+    paramSchema = resolveSchema(getDefaultContext(options, components), _.cloneDeep(param.schema),
+      0, PROCESSING_TYPE.VALIDATION),
+    isEvenNumber = (num) => {
+      return (num % 2 === 0);
+    },
+    convertToDataType = (value) => {
+      try {
+        return JSON.parse(value);
+      }
+      catch (e) {
+        return value;
+      }
+    };
+
+  // for invalid param object return null
+  if (!_.isObject(param) || !_.isString(paramValue)) {
+    return null;
+  }
+
+  let { startValue, propSeparator, keyValueSeparator, isExplodable } =
+    getParamSerialisationInfo(param, parameterSource, components, options);
+
+  // as query params are constructed from url, during conversion we use decodeURI which converts ('%20' into ' ')
+  (keyValueSeparator === '%20') && (keyValueSeparator = ' ');
```

Comment:
@barshan23 Because this information is used for both `serializing` and `deserialising`. We can use space directly for deserialising but for serialising we use `%20`.